src/actions/actions.js

```diff
@@ -0,0 +1,231 @@
+import {
+  START_FETCHING_IMAGE,
+  FETCHING_IMAGE_ERROR,
+  START_FETCH_TAGS,
+  FETCHING_TAGS_ERROR,
+  SUBMIT_TAGS_START,
+  SUBMIT_TAGS_SUCCESS,
+  SUBMIT_TAGS_ERROR,
+  FETCH_IMAGE_SUCCESS,
+  FETCH_TAG_SUCCESS,
+  TOGGLE_MODE,
+  SELECT_IMAGE,
+  UPDATE_SCORE,
+  RESET_SCORE,
+  RESET_GRID,
+  CHANGE_GRID,
+  RESET_ERROR,
+  VIEW_TAGS,
+  VIEW_IMAGE,
+  SEARCH_TAG_NAME,
+  RESET_TAG_SEARCH,
+  SAVING_IMAGE_START,
+} from './actionTypes';
+
+export function retrieveImageIndex() {
+  return async function getImages(dispatch) {
+    dispatch(startImageFetch());
+    await fetch('http://localhost:3001', { mode: 'cors' })
+      .then(async function(response) {
```

Comment:
generally if we use `async`/`await`, we don't need to use Promise utilities (eg, `.then`, `.catch`, etc) and vice versa. there will be times when we may need to use both but these simple data fetches isn't one of those scenarios. it is useful to know that they *can* be combined if needed and that `async`/`await` is basically syntactic sugar over Promises.

if using `async`/`await`:

```js
export function retrieveImageIndex() {
  return async function getImages(dispatch) {
    dispatch(startFetch());

    try {
      const response = await fetch(...);

      if (response.status === 200) {
        let data = await response.json();
        dispatch(fetchSuccess(data));
      } else {
        // do error stuff
        dispatch(fetchError(error));
      }
    } catch(error) {
      dispatch(fetchError(error));
    }
  }
}
```

if using promises:

```js
export function retrieveImageIndex() {
  return async function getImages(dispatch) {
    dispatch(startFetch());

    return fetch(...);
      .then((response) => {
        if (response.status === 200) {
          return response.json().then(data => fetchSuccess(data));
        } else {
          // error stuff
          dispatch(fetchError(...));
          // return or throw, depends on error handling strategies you want to use in your app
        }
      })
      .catch(error) => dispatch(fetchError(error));
  }
}
```

Reply:
Ah okay, I was getting an error from React saying I wasn't handling fetch errors. I guess I was supposed to do try/catch, not then/catch. Forgot about try lol